sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalMathUtils.scala

```diff
@@ -31,16 +31,17 @@ object IntervalMathUtils {
 
   def subtractExact(a: Long, b: Long): Long = withOverflow(Math.subtractExact(a, b), "try_subtract")
 
-  def negateExact(a: Int): Int = withOverflow(Math.negateExact(a))
+  def negateExact(a: Int): Int = withOverflow(Math.negateExact(a), "try_multiply")
```

Comment:
I think it's better to suggest `try_multiply` here instead of suggesting nothing. Hope user will find out that `-x` could be replaced by `try_multiply(x, -1)`.

Reply:
Unfortunately this is not a safe thing to do. UnaryMinus is introduced in analysis by many different functions and we cannot guarantee that the error that occurred here is from -x or from some expression that internally converts x to -x.

Reply:
Ok, I'll revert this change.
But why we can guarantee hints for subtractExact or multiplyExact? They aren't used in analysis?